### DISCUSSION TITLE
Variable use in rule trigger

### DISCUSSION THREAD

**Discussion by bwmtx** [arendst/Tasmota#15678] [Q&A] [answered]
Is the below possible?  I can't get it to work, not sure if I'm just doing something wrong or if it's not possible.

mem1 3
mem2 100

Rule1 ON DS18B20-%mem1%#temperature>%mem2% DO  publish tele/ds18b20-%mem1%/SENSOR DANGER ENDON

**Comment by barbudor** [comment, marked as answer]
As far as I can see in the code, variable replacement is performed only after the last  # so there couldn't be a variable in the trigger name.

What would be your use-case to do so ?
By compiling your own build with `#define DS18x20_USE_ID_AS_NAME` in your `user_config_override.h` , you will get an fixed ID based on the last 3 bytes chip's internal ID

**Reply by bwmtx** [reply]
All good.  After some thought, I don't need the sensor index to be dynamic.  Just need to be able to compare the temp to a variable, which works.
